### PR TITLE
KAZOO-4417 xmpp new library and some fixes

### PR DESCRIPTION
**Pull request by lazedo**
and some fax fixes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ before_install:
 script: ERLC_OPTS="$ERLC_OPTS" make compile
 
 after_success:
-  - if [[ 0 -ne `git status --porcelain -- applications/ core/ deps/ | wc -l` ]]; then echo Unstaged changes! && git status --porcelain && exit 1; fi
```
Comment (fenollp) [inline .travis.yml]: Vetoooooo!

Comment (fenollp) [inline .travis.yml]: Actually no veto. That should have been removed after `.gitignore` sanitization.

File: applications/fax/src/wapi_xmpp.erl [hunk 2]
```diff
@@ -94,3 +99,19 @@ publish_event(Event, ContentType) ->
...
+    regexp_get(JID, <<"^([^/]*)">>).
+
+jid_username(JID) ->
+    regexp_get(JID, <<"^([^@]*)">>).
+
+jid_server(JID) ->
+    regexp_get(JID, <<"^[^@]*[@]([^/]*)">>).
+
+jid_resource(JID) ->
+    regexp_get(JID, <<"^[^/]*[/](.*)">>).
```
Comment (fenollp) [inline applications/fax/src/wapi_xmpp.erl]: Do you have functional tests for these?

Comment (lazedo) [inline applications/fax/src/wapi_xmpp.erl]: JID is always user@domain/resource

Comment (fenollp) [inline applications/fax/src/wapi_xmpp.erl]: Why use regexps then?
```erlang
[User,Domain,Resource] = binary:split(JID, [<<"@">>, <<"/">>], ['global'])
```

Comment (lazedo) [inline applications/fax/src/wapi_xmpp.erl]: its an option, does it not work ?

**Comment by fenollp** [comment]
@lazedo What's the repos you got the deps from? I'd like to keep tabs on these for easier updates in the future

**Comment by lazedo** [comment]
@fenollp https://github.com/esl/escalus as base

**Pull request merged**